### PR TITLE
feat: Add support for advanced redirect types

### PR DESCRIPTION
**Pull request by sarahetter**
This PR adds support for advanced redirect types in gatsby, as mentioned in their [support blog post](https://support.gatsbyjs.com/hc/en-us/articles/1500003051241-Working-with-Redirects)
* Wildcard path redirects
* Splat redirects
* Country based redirects
* Language based redirects
* Query param redirects

I've also added a snapshot test to ensure that each of these types of redirects are being output to the _redirects file properly.

Closes: #93

### CODE REVIEW

**Review by ascorbic — APPROVED** [review]
Nice work! I didn't realise they'd done a blog about it. We should post something about this, even if it's just an announcement in the forums

File: src/create-redirects.ts [hunk 1]
```diff
@@ -17,11 +29,15 @@ export default async function writeRedirectsFile(pluginData: any, redirects: any
     `headers`,
     `signed`,
     `edge_handler`,
-    `Language`,
-    `Country`,
```
Comment (jaska120) [inline src/create-redirects.ts:21]: This was a **breaking change** and it should have been introduced in a major bump. All the redirects were possible previously, but the change broke our delpoyment. I do agree that the `conditions` key may be more suitable, but let's just be cautious with the breaking changes in the future.

Comment (jaska120) [inline src/create-redirects.ts:21]: This was a **breaking change** and it should have been introduced in a major bump. All the redirects were possible previously, but the change broke our delpoyment. I do agree that the `conditions` key may be more suitable, but let's just be cautious with the breaking changes in the future.

Comment (jaska120) [reply src/create-redirects.ts:21]: @ascorbic

**Pull request merged**